cpp/src/Slice/Grammar.y

```diff
@@ -206,6 +206,14 @@ file_metadata
 }
 ;
 
+// ----------------------------------------------------------------------
+local_metadata
+// ----------------------------------------------------------------------
+: ICE_METADATA_OPEN metadata_list ICE_METADATA_CLOSE
+{
+    $$ = $2;
+}
+
```

Comment:
We were matching these tokens in the `metadata_directives` rule you'll see befow this one.
I just refactored them into a separate rule for grammar reasons.